Update CHANGELOG.md for v5.9.0 (#16752)
hashicorp/terraform-provider-google: CHANGELOG.md

## 5.9.0 (Unreleased)

FEATURES:
* **New Data Source:** `google_logging_folder_settings` ([#16658](https://github.com/hashicorp/terraform-provider-google/pull/16658))
* **New Data Source:** `google_logging_organization_settings` ([#16658](https://github.com/hashicorp/terraform-provider-google/pull/16658))
* **New Data Source:** `google_logging_project_settings` ([#16658](https://github.com/hashicorp/terraform-provider-google/pull/16658))
* **New Data Source:** `google_vmwareengine_network_policy` ([#16639](https://github.com/hashicorp/terraform-provider-google/pull/16639))
* **New Data Source:** `google_vmwareengine_nsx_credentials` ([#16669](https://github.com/hashicorp/terraform-provider-google/pull/16669))
* **New Resource:** `google_scc_event_threat_detection_custom_module` ([#16649](https://github.com/hashicorp/terraform-provider-google/pull/16649))
* **New Resource:** `google_secure_source_manager_instance` ([#16637](https://github.com/hashicorp/terraform-provider-google/pull/16637))
* **New Resource:** `google_vmwareengine_network_policy` ([#16639](https://github.com/hashicorp/terraform-provider-google/pull/16639))

IMPROVEMENTS:

> * bigqueryconnection - add `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))

Reviewer: ```suggestion
* bigqueryconnection - added `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))
```

> * bigqueryconnection - added `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))

Author: ```suggestion
* bigqueryconnection: added `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))
```

* cloudidentity: added `expiry_detail` field to `google_cloud_identity_group_membership` resource ([#16643](https://github.com/hashicorp/terraform-provider-google/pull/16643))
* container: added `queued_provisioning` field to `google_container_node_pool` resource ([#16623](https://github.com/hashicorp/terraform-provider-google/pull/16623))

> * container: promoted GKE Autoscaling Profiles to GA in the `cluster_autoscaling` block in `google_container_cluster` resource ([#16653](https://github.com/hashicorp/terraform-provider-google/pull/16653))

Reviewer: ```suggestion
* container: promoted `autoscaling_profile` to GA in the `cluster_autoscaling` block in `google_container_cluster` resource ([#16653](https://github.com/hashicorp/terraform-provider-google/pull/16653))
```

> * dlp: update required fields in `timespan_config` config block for `data_loss_prevention_job_trigger` resource ([#16628](https://github.com/hashicorp/terraform-provider-google/pull/16628))

Reviewer: ```suggestion
* dlp: updated required fields in `timespan_config` config block for `data_loss_prevention_job_trigger` resource ([#16628](https://github.com/hashicorp/terraform-provider-google/pull/16628))
```

* gkehub: added `default_cluster_config` field to `google_gke_hub_fleet` resource  ([#16630](https://github.com/hashicorp/terraform-provider-google/pull/16630))
* gkehub: added `binary_authorization_config` field to `google_gke_hub_fleet` resource ([#16674](https://github.com/hashicorp/terraform-provider-google/pull/16674))
* sql: added support for in-place updates to the `edition` field in `google_sql_database_instance` resource ([#16629](https://github.com/hashicorp/terraform-provider-google/pull/16629))

BUG FIXES:
* artifactregistry: fixed permadiff due to unsorted `virtual_repository_config` array in `google_artifact_registry_repository` ([#16646](https://github.com/hashicorp/terraform-provider-google/pull/16646))
* container: made `dns_config` field updatable on `google_container_cluster` resource ([#16652](https://github.com/hashicorp/terraform-provider-google/pull/16652))
* firestore: retried resource creation for error 409 with the text "Aborted due to cross-transaction contention" in `google_firestore_index` resource ([#16618](https://github.com/hashicorp/terraform-provider-google/pull/16618))

> * firestore: retried resource creation for error 409 with the text "Please retry, underlying data changed" in `google_firestore_index` resource ([#16670](https://github.com/hashicorp/terraform-provider-google/pull/16670))

Author: They're for different text in the error message - should I still consolidate them?

> * firestore: retried resource creation for error 409 with the text "Please retry, underlying data changed" in `google_firestore_index` resource ([#16670](https://github.com/hashicorp/terraform-provider-google/pull/16670))

Reviewer 2: I think so. It's not clear to me on initial reading what the differences is. What might a user think reading these notes.
